drivers/SmartThings/philips-hue/src/lunchbox/sse/eventsource.lua

```diff
@@ -156,8 +156,8 @@ local valid_fields = util.read_only {
 -- h/t to github.com/FreeMasen for the suggestions on the efficient implementation of this
 local function find_line_endings(chunk)
   local r_idx, n_idx = string.find(chunk, "[\r\n]+")
-  if r_idx == n_idx then
-    -- 1 character
+  if r_idx == nil or r_idx == n_idx then
+    -- 1 character or no match
```

Comment:
When is the no match happening?

Reply:
For a well-formed message, it shouldn't, but for any kind of adversarial input, someone could hand off a string that doesn't use CRLF.

For the specific case of the EventSource it *shouldn't* happen unless the payload gets corrupted.

But from the perspective of type checking, it's possible for `find` to return `nil` on no match, so we handle it.